### PR TITLE
backports for 0.5.9 -- small and/or important fixes

### PR DESCRIPTION
**Pull request by chappjc**
We're prepping a v0.5.9 release because we need a tag with the new Docker build actions, and there are a number of important fixes since ~December 2022 that are worth cherry-picking.
Since 0.6 is coming very soon, this wasn't planned.

### CODE REVIEW

**Comment by chappjc** [comment]
Anything you guys think we should add or remove? @buck54321 @martonp @JoeGruffins 

The only things that would be really optional are that DB compaction commit and the estimateSwap fix commit.  The rest are pretty valuable fixes if we're rolling one last v0.5 patch relese.

**Review by JoeGruffins — COMMENTED** [review]
web has the json thing:
```
$ npm audit
# npm audit report

json5  <1.0.2 || >=2.0.0 <2.2.2
Severity: high
Prototype Pollution in JSON5 via Parse Method - https://github.com/advisories/GHSA-9c47-m6qq-7p4h
Prototype Pollution in JSON5 via Parse Method - https://github.com/advisories/GHSA-9c47-m6qq-7p4h
fix available via `npm audit fix`
node_modules/json5
node_modules/tsconfig-paths/node_modules/json5

loader-utils  2.0.0 - 2.0.3
Severity: critical
loader-utils is vulnerable to Regular Expression Denial of Service (ReDoS) via url variable - https://github.com/advisories/GHSA-3rfm-jhwj-7488
loader-utils is vulnerable to Regular Expression Denial of Service (ReDoS) - https://github.com/advisories/GHSA-hhq3-ff78-jv3g
Prototype pollution in webpack loader-utils - https://github.com/advisories/GHSA-76p3-8jx3-jpfq
fix available via `npm audit fix`
node_modules/loader-utils

2 vulnerabilities (1 high, 1 critical)
```

**Comment by JoeGruffins** [comment]
Trying to use a db I was using on master on testnet, get an error when trying to add a dex. May be normal. Will start anew and see if its fine:
```
2023-02-22 03:31:07.521 [DBG] WEB[WS]: New websocket client 127.0.0.1:57058
2023-02-22 03:31:09.343 [INF] CORE: Connecting wallet for dcr
2023-02-22 03:31:09.936 [INF] CORE: Connected to 1 of 1 wallets.
2023-02-22 03:31:12.937 [INF] CORE: Wallet synced for asset dcr
2023-02-22 03:31:31.223 [DBG] CORE: Setting server api version to 0.
2023-02-22 03:31:31.223 [DBG] CORE: Server 127.0.0.1:17273 broadcast timeout 5m0s. Tick interval 37.5s
2023-02-22 03:31:31.223 [INF] CORE: Connected to DEX server at 127.0.0.1:17273 but NOT listening for messages.
2023-02-22 03:31:31.233 [DBG] CORE: Authenticated connection to 127.0.0.1:17273, acct 1aa4aec190d355ead10b956fed232f8329a3234a2062e467b97e1b2727c291aa, 0 active orders, 0 active matches, score 0 (suspended = false)
2023-02-22 03:31:31.233 [DBG] CORE[wss://127.0.0.1:17273/ws]: Sending close 1000 (normal) message.
2023-02-22 03:31:31.233 [ERR] WEB: error saving restored account: failed to create account bucket
```

**Comment by chappjc** [comment]
The AccountInfo has a v3 encoding on master so I think that's expected if you try to go back.  A little unfortunate that we're throwing out the actual error in `CreateAccount` though.

**Comment by chappjc** [comment]
Oh, I think if you look back in your logs you'll see an ignored error. Related: https://github.com/decred/dcrdex/pull/2119
The thrown away error from that line in `CreateAccount` was almost certainly `bucket already exists`, because it simply failed to load on `initialize`.  We should merge https://github.com/decred/dcrdex/pull/2119
I had hit the same thing

**Comment by JoeGruffins** [comment]
Oh, looking up I see `2023-02-22 03:30:47.018 [ERR] CORE: Error retrieving accounts from database: unknown AccountInfo version 3`

**Comment by chappjc** [comment]
OK, that's covered.  Shoulda merged that 2 wks ago.  Sigh...

**Comment by chappjc** [comment]
I'm almost certain those weren't runtime vulnerabilities, but updated anyway.  Had to checking a new webpack js bundle (client/webserver/site/dist/entry.js) and update the html cache busters in the templates.  Best to recheck the UI after that.

**Comment by JoeGruffins** [comment]
Ok, looking around.

File: client/webserver/site/dist/entry.js [hunk 1]
```diff
@@ -1,2 +1,2 @@
...se 6:case"end":return e.stop()}}),e,this)}))),function(e){return n.apply(this,arguments)})},{key:"checkResponse",value:function(e,t){return!(!e.requestSuccessful||!e.ok)||(this.user.inited&&!t&&this.notify(Xt(ke(ae),e.msg,5)),!1)}},{key:"signOut",value:(t=o(y().mark((function e(){var t;return y().wrap((function(e){for(;;)switch(e.prev=e.next){case 0:return e.next=2,nt("/api/logout");case 2:if(t=e.sent,this.checkResponse(t)){e.next=6;break}return De.hide(this.page.profileBox),e.abrupt("return");case 6:Ve.clearAllStore(),Ve.removeAuthCK(),window.location.href="/login";case 9:case"end":return e.stop()}}),e,this)}))),function(){return t.apply(this,arguments)})}]),e}(),Nr=(u(Fr={},3,"good"),u(Fr,5,"bad"),u(Fr,4,"warn"),Fr);function Ur(e){return e.replace(/^\//,"").split("/")[0].split("?")[0].split("#")[0]}function zr(){var e=new Date,t=e.getHours().toString().padStart(2,"0"),n=e.getMinutes().toString().padStart(2,"0"),r=e.getSeconds().toString().padStart(2,"0"),a=e.getMilliseconds().toString().padStart(3,"0");return"".concat(t,":").concat(n,":").concat(r,".").concat(a)}function jr(e,t){e.classList.remove("bad","warn","good"),e.classList.add(Nr[t])}var _r=new Mr;Je=_r,_r.start()})()})();
```
Comment (chappjc) [inline client/webserver/site/dist/entry.js:1]: Only change I see in the bundle is `this.commitHash="fb5d9ffd9fc16487f659f6dc99e32e43c0aa2711"`, so yeah, the npm fix was not a runtime vuln.  Probably not even a vuln for dev, but done anyway.

Comment (chappjc) [reply client/webserver/site/dist/entry.js:1]: New bundle still working fine for me.
Have had this branch on mainnet all day, just without the last three commits, until now.

**Review by JoeGruffins — APPROVED** [review]

**Comment by buck54321** [comment]
Test well upgrading against 0.5.8.
I did notice that the 0.5 release still allows creation of native eth wallets, which is a little weird, but meh. Looking forward to 0.6.

**Review by buck54321 — APPROVED** [review]

**Pull request merged**